assets/js/main.js

```diff
@@ -0,0 +1,95 @@
+$(function() {
+  function slider() {
+    const $slides = $('.slides');
+    const $slide = $('.slide');
+    const $indicator = $('.indicator');
+    const $next = $('.next');
+    const $prev = $('.prev');
+    let index = 0;
+    let indicatorItem = '';
+    let flag = 0;
+
+    function addIndicator() {
+      for (let i = 0; i < $slide.length; i++) {
+        $indicator.append('<li class="indicator-item"></li>');
+      }
+      indicatorItem = $('.indicator-item');
+      indicatorItem.eq(index).addClass('is-active');
+    }
+
+    function updateNav() {
+      indicatorItem.removeClass('is-active').eq(index).addClass('is-active')
+    }
+
+    function changeSlide() {
+      $slides.animate({
+        left: index * -100 + '%'
+      }, 1000);
+      if(index == $slide.length){
+        index = 0;
+        $slides.animate({
+          left: index * -100 + '%'
+        }, 0);
+      }else if(index == -1) {
+        index = $slide.length -1;
+        $slides.animate({
+          left: index *-100 + '%'
+        }, 0);
+      }
+      updateNav();
+      console.log(index)
+    }
+
+    function prevSlide() {
+      if (flag == 0) {
+        flag = 1;
+        index--;
+        changeSlide();
+        setTimeout(function() {
+          flag = 0
+        }, 1000);
+      }
+    }
+
+    function nextSlide() {
+      if (flag == 0) {
+        flag = 1;
+        index++;
+        changeSlide();
+        setTimeout(function() {
+          flag = 0
+        }, 1000);
+      }
+    }
+
+    function indicatorSlide() {
+      if (flag == 0) {
+        flag = 1;
+        index = $(this).index();
+        changeSlide();
+        setTimeout(function() {
+          flag = 0
+        }, 1000);
+      }
+    }
+
+    function autoSlide() {
+      const interval = 1000;
+      setInterval(nextSlide,interval);
+    }
+
+    function slideEvent() {
+      autoSlide()
+      $prev.on('click', prevSlide);
+      $next.on('click', nextSlide);
+      indicatorItem.on('click', indicatorSlide);
+    }
+
+    function init() {
+      addIndicator();
+      slideEvent();
+    }
```

Comment:
ここはsetEventにして、domへのイベントをまとめる関数にした方が良いです

```suggestion
    function setEvent() {
      $prev.on('click', prevSlide);
      $next.on('click', nextSlide);
      indicatorItem.on('click', indicatorSlide);
    }

    function init() {
      addIndicator();
      autoSlide()
      setEvent();
    }
```

Reply:
fed95e8
修正しました